gdb-server/src/worker.rs

```diff
@@ -19,7 +19,7 @@ type Receiver<T> = mpsc::UnboundedReceiver<T>;
 pub async fn worker(
     mut input_stream: Receiver<CheckedPacket>,
     output_stream: Sender<CheckedPacket>,
-    session: &mut Session,
+    session: Arc<Mutex<Session>>,
```

Comment:
```suggestion
    session: &Mutex<Session>,
```
All other usages of `Arc<Mutex<Session>>` in this file could also be replaced, it is not necessary to carry the `Arc` around everywhere, as we never actually need to clone it here.